core/ledger/kvledger/snapshot.go

```diff
@@ -371,3 +384,28 @@ func verifyFileHash(dir, file string, expectedHashInHex string, hashProvider led
 	}
 	return nil
 }
+
+type mostRecentCollectionConfigFetcher struct {
+	*confighistory.Retriever
+	ledger.DeployedChaincodeInfoProvider
+}
+
+func (c *mostRecentCollectionConfigFetcher) CollectionInfo(chaincodeName, collectionName string) (*peer.StaticCollectionConfig, error) {
+	isImplicitCollection, mspID := implicitcollection.MspIDIfImplicitCollection(collectionName)
+	if isImplicitCollection {
+		return c.GenerateImplicitCollectionForOrg(mspID), nil
+	}
+
+	explicitCollections, err := c.MostRecentCollectionConfigBelow(math.MaxUint64, chaincodeName)
+	if err != nil || explicitCollections == nil || explicitCollections.CollectionConfig == nil {
+		return nil, errors.WithMessage(err, "error while fetching most recent collection config")
+	}
```

Comment:
Maybe we can return `nil` if the collectionConfig is `nil`? Anyway, `GetBTL()` does not expect an error for a `nil` collectionConfig https://github.com/hyperledger/fabric/blob/bbea1f582f360f2557314dd2aa3abb5c8e33a4b8/core/ledger/pvtdatapolicy/btlpolicy.go#L58-L64

Reply:
That's exactly what this code is doing

Reply:
Ah. Yeah, I was mistaken. The second/third time I got confused due to this pattern
```
	if err != nil || explicitCollections == nil || explicitCollections.CollectionConfig == nil {
		return nil, errors.WithMessage(err, "error while fetching most recent collection config")
	} 
```
where the err can be nil here.